jina/types/document/multimodal.py

```diff
@@ -0,0 +1,44 @@
+from typing import List
+
+from . import Document
+from ...excepts import LengthMismatchException, BadDocType
+
+class MultimodalDocument(Document):
+    """Each :class:`MultimodalDocument` should have at least 2 chunks (represent as :class:`DocumentSet`)
+    and len(set(doc.chunks.modality)) == len(doc.chunk)
+    """
+    def __init__(self, document = None,
+                 copy: bool = False, **kwargs):
+        super().__init__(document=document, copy=copy, **kwargs)
+        self._modalities = set()
+        self._doc_content = {}
+        if len(self._document.chunks) < 2:
+            raise BadDocType(
+                f'A MultimodalDocument should have at least 2 modalities, {len(self.document.chunks)} received.')
+        if len(self._document.chunks) != len(self.modalities):
+            raise LengthMismatchException(
+                f'Document has {len(self._document.chunks)} chunks and {len(self.modalities)} modalities')
+
+    def extract_content(self) -> List[str]:
```

Comment:
just to make sure it is understood, the original code is returning a `Dict` and not a `List` (maybe it is better a List) (Just raising awareness)

Reply:
you're right, I updated the annotation